File: Ander systeem/Core.ApplicationServices/DTO/Calculations/CalculationDTO.cs
using System.Text.Json.Serialization;

namespace Leap.ApplicationServices.DTO.Calculations
{
    [JsonDerivedType(typeof(KPIDTO), typeDiscriminator: "KPIDTO")]
    public class CalculationDTO
    {
        public int CalculationId { get; set; }
        public required string OutputColumn { get; set; }
        public required IEnumerable<string> InputColumns { get; set; } = new List<string>();
    }

    public class KPIDTO : CalculationDTO
    {
        public required string CalculationString { get; set; } // Maps to CalculationString
        public required IEnumerable<string> OperationsList { get; set; } // Maps to ConcatCalculationString
    }
}

File: App Python/Domain/DTO/CalculationDTO.py
from typing import Union

from pydantic import BaseModel


class CalculationDTO(BaseModel):
    InputColumns: list[str] = []
    OutputColumn: str

class KPIDTO(CalculationDTO):
    CalculationString: str 
    OperationsList: list[str]

# class AggregationDTO(CalculationDTO):
#     Action: str 

CalculationDTOUnion = Union[KPIDTO]


